python/eos/analysis.py

```diff
@@ -400,19 +400,29 @@ def sample_pmc(self, log_proposal, step_N=1000, steps=10, final_N=5000, rng=np.r
             generating_components.append(origins)
             samples = sampler.samples[:]
             last_weights = np.copy(sampler.weights[-1][:, 0])
-            for i, w in enumerate(last_weights):
-                if w <= 0 or np.isnan(w):
-                    last_weights[i] = eps
-            normalized_last_weights = last_weights / np.sum(last_weights)
+            normalized_last_weights = last_weights / np.sum(
```

Comment:
This fails if none of the weights is positive, which is a pathological corner case. Should we test for it and skip the computation of the perplexity in that case?

Reply:
Yes, I set perplexity to zero in this case